dvclive/report.py

```diff
@@ -0,0 +1,56 @@
+import os
+import subprocess
+from pathlib import Path
+from tempfile import TemporaryDirectory
+
+from dvclive.utils import _parse_tsv, is_cml_available
+
+
+def build_report(plots_folder):
```

Comment:
It seems more generally useful for `dvc plots show` (or wherever we extract the plots rendering library) to have a markdown format option. This would also make it easier in general to generate reports in CML instead of needing to use tools like vl2png. Not sure how we can also include metrics or if that should stay separate 🤔 . cc @pared

Reply:
I think that ideally we should have an option to define a report both for both DVC and `dvclive` - including metrics plots and maybe even params - though that would require some research - including metrics and params as a markdown table seems doable, I don't know about vega plots - how to handle image generation, which currently leverages HTML and `<script>`.

Reply:
Hmm, looks like markdown plots generation might require altair and additional packages to do in vega-lite: https://altair-viz.github.io/user_guide/saving_charts.html#png-svg-and-pdf-format. Alternatives would be to have other rendering options like matplotlib (what @daavoo uses here) or plotly (which looks to be the most lightweight option).

Re: including metrics and maybe params, I'm not sure whether this belongs in dvc, dvclive, or both. I think the current table makes sense for dvclive since it only knows about those metrics, and dvc doesn't know about steps. I could see value in including metrics, params, and plots altogether in dvc outputs, but this might have to be separate issue.

Reply:
> It seems more generally useful for dvc plots show (or wherever we extract the plots rendering library) to have a markdown format option.

I see 2 different scopes. 

On the one hand, this P.R. focuses on monitoring a running stage only covering the DVCLive scope (current workspace and metrics and plots logged by `Live`). 

On the other, it seems to me that the discussion is more on the lines of extending/simplifying the existing [CML/DVC integration](https://cml.dev/doc/cml-with-dvc) which happens after a stage has been finished.

IMO, having `--md` options in `dvc metrics`, `dvc params` and `dvc exp` commands is better than trying to join all together in a single output. It allows users to structure their reports on demand. 

Adding `--md` option to `dvc plots` command would definitely smooth the CML integration. It's not clear how to do with the current vega-based rendering but it seems doable if we extend the rendering options.